spec.bs

```diff
@@ -1600,6 +1631,16 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
       [=interest group/owner=].
   1. Run [=report win=] with |leadingBidInfo|, |sellerSignals|, |reportResultBrowserSignals|, and
     |directFromSellerSignalsForWinner|.
+1. Let |replacements| be an [=ordered map=] whose [=map/keys=] are [=strings=] and whose [=map/values=]
+    are [=strings=].
+  1. [=list/For each=] [=ad keyword replacement=], |replacement|, within [=auction config/deprecated render url replacements=]:
+    1. Let |k| be |replacement|'s [=ad keyword replacement/match=].
+    1. Let |v| be |replacement|'s [=ad keyword replacement/replacement=].
+    1. [=map/Set=] |replacements|[|k|] to |v|.
+1. Set |leadingBidInfo|'s [=leading bid info/leading bid=]'s [=generated bid/ad descriptor=] to the result of [=fencedframeutil/substitute macros=] with |replacements| and [=leading bid info/leading bid=]'s [=generated bid/ad descriptor=].

```

Comment:
after reading your CL more carefully and confirmed with Russ, that the ad URL(s) are only replaced for ad rendering. K-anon and reportWin/reportResult both get original un-replaced URLs. So we cannot change |leadingBidInfo|'s ad descriptor(s), which will affect all of them. I had a wrong assumption that all these places would have replaced URLs.

So instead, we need to do this inside [[=fill in a pending fenced frame config=]](https://wicg.github.io/turtledove/#fill-in-a-pending-fenced-frame-config). Step 2 and 10 of the algorithm uses ad descriptor and ad descriptors. You need to have two new variables for the two's macros replaced, and "Set pendingConfig’s mapped url's value" to the new variables instead of the |winningBid|'s original ad fields.

Reply:
Thanks!